File: library-frontend/src/services/BookService.js
import api from "../api/axoisConfig";

const BOOK_URL = "/books";

const BookService = {

    getAllBooks() {

        return api.get(BOOK_URL);

    },

    getBookById(id) {

        return api.get(`${BOOK_URL}/${id}`);

    },

    addBook(book) {

        return api.post(BOOK_URL, book);

    },

    updateBook(id, book) {

        return api.put(
            `${BOOK_URL}/${id}`,
            book
        );

    },

    deleteBook(id) {

        return api.delete(
            `${BOOK_URL}/${id}`
        );

    }

};

export default BookService;

File: library-app/src/main/java/com/veerdev/service/BookService.java
package com.veerdev.service;

import java.util.List;

import com.veerdev.entity.Book;

public interface BookService {

	 Book addBook(Book book);

	    List<Book> getAllBooks();

	    Book getBookById(Integer id);

	    Book updateBook(Integer id, Book book);

	    void deleteBook(Integer id);
	    
}
